### ISSUE TITLE
README is unclear

### ISSUE THREAD

**Issue by timholy** [ZacLN/SparseInterp.jl#1]
I'm now guessing this package refers to https://en.wikipedia.org/wiki/Sparse_grid, but I had to dig into the source and find the word "Smolyak" to discover that. Overall the purpose and usage of the package is a little confusing. (Examples: what does "fineness" mean and how does one specify it? "`x` the points to be evaluated": more than one N-dimensional point? If so is it rows or columns of `x`? etc.)

**Comment by ZacLN** [comment]
I'll write up a readme, this is an old one!

**Comment by timholy** [comment]
I should have clarified that I found this because you're registering it. A spotlight tends to attract curious visitors who then demand awesomeness :smile:. It looks potentially interesting, I haven't dug into the math enough yet to understand how it improves quadrature, but the next time this issue comes up I'll be back.

**Comment by stevengj** [comment]
I know a (very) little about smolyak rules, and I'm still pretty unclear on this package.  In particular, what quadrature/interpolation scheme is it using?

I plotted some of the `values(NGrid(...))` output in 2d, and the points don't look like the ones you would get from e.g. Chebyshev or Gauss–Patterson rules.  Your points look very regularly spaced, making me think that you are using a low-order polynomial (analogous to a trapezoidal rule with equally spaced points) or something.  e.g. compare your grid for `values(NGrid([8,8]))`
![image](https://user-images.githubusercontent.com/2913679/35991087-086a33bc-0cd4-11e8-8b0d-dff9c1cdd54d.png)
to [a Chebyshev–Smolyak grid](https://robertdkirkby.github.io/IntroToChebyshevSmolyak.html)
![image](https://robertdkirkby.github.io/IntroToChebyshevSmolyak_05.png)

**Comment by stevengj** [comment]
Also, how does this package compare to https://github.com/ABAtanasov/GalerkinSparseGrids.jl?

**Comment by ZacLN** [comment]
You are correct, it uses an equally spaced grid defined [here](https://github.com/ZacLN/SparseInterp.jl/blob/master/src/univariategrids.jl#L4) and corresponding linear hat basis functions, though other univariate grids can be used. I've got a paper that describes the algorithm somewhere but I'll have to hunt around for it.
